org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/general/completion/FilePathCompletionParticipant.java

```diff
@@ -132,6 +132,9 @@ private static void addCompletionItems(String value, ICompletionRequest request,
 					return;
 				}
 			}
+		} else if (!hasPathBeginning(valuePath)) {

```

Comment:
This logic is very annoying with the case of attribute that we are sure that it is a path like
`<!DOCTYPE foo SYSTEM "|">` must open the file path completion.

Reply:
I looked into the code, and this case would already not return completion, since it's an empty string. I've changed that, so that in doctypes, when you're within `""`, you always get filepath completion. My implementation is pretty naive, but luckily, paths are not suggested when you attempt completion in the subset, eg.

```xml
<!DOCTYPE aaa [
    <!ELEMENT aaa (#PCDATA)>
    <!ENTITY entity-name "|">
]>
<aaa Name=""></aaa>
```
There is no file completion returned at `|`.